source/concepts.tex

```diff
@@ -0,0 +1,1249 @@
+type}, \tcode{C}.
+\begin{note}
+\tcode{C} could be the same as \tcode{T}, or \tcode{U}, or it could be a different
+type. \tcode{C} may be a reference type.
+\end{note}
+
+\indexlibrary{\idxcode{CommonReference}}%
+\begin{itemdecl}
+template <class T, class U>
+concept CommonReference =
+  Same<common_reference_t<T, U>, common_reference_t<U, T>> &&
+  ConvertibleTo<T, common_reference_t<T, U>> &&
+  ConvertibleTo<U, common_reference_t<T, U>>;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{C} be \tcode{common_reference_t<T, U>}. Let \tcode{t} be a
+function whose return type is \tcode{T}, and let \tcode{u} be a function
+whose return type is \tcode{U}. \tcode{CommonReference<T, U>} is satisfied
+only if:
+\begin{itemize}
+\item \tcode{C(t())} equals \tcode{C(t())} if and only if \tcode{t()} is an
+  equality preserving expression~(\ref{concepts.lib.general.equality}).
+\item \tcode{C(u())} equals \tcode{C(u())} if and only if \tcode{u()} is an
+  equality preserving expression.
+\end{itemize}
+
+\pnum
+\begin{note}
+Users can customize the behavior of \tcode{CommonReference} by specializing the
+\tcode{basic_common_reference} class template~(\ref{meta.trans.other}).
+\end{note}
+\end{itemdescr}
+
+\rSec2[concepts.lib.corelang.common]{Concept \tcode{Common}}
+
+\pnum
+If \tcode{T} and \tcode{U} can both be explicitly converted to some third type,
+\tcode{C}, then \tcode{T} and \tcode{U} share a \term{common type},
+\tcode{C}.
+\begin{note}
+\tcode{C} could be the same as \tcode{T}, or \tcode{U}, or it could be a different
+type. \tcode{C} might not be unique.
+\end{note}
+
+\indexlibrary{\idxcode{Common}}%
+\begin{itemdecl}
+template <class T, class U>
+concept Common =
+  Same<common_type_t<T, U>, common_type_t<U, T>> &&
+  ConvertibleTo<T, common_type_t<T, U>> &&
+  ConvertibleTo<U, common_type_t<T, U>> &&
+  CommonReference<
+    add_lvalue_reference_t<const T>,
+    add_lvalue_reference_t<const U>> &&
+  CommonReference<
+    add_lvalue_reference_t<common_type_t<T, U>>,
+    common_reference_t<
+      add_lvalue_reference_t<const T>,
+      add_lvalue_reference_t<const U>>>;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{C} be \tcode{common_type_t<T, U>}. Let
+\tcode{t} be a function whose return type is \tcode{T}, and let \tcode{u} be a function
+whose return type is \tcode{U}. \tcode{Common<T, U>} is satisfied only if:
+\begin{itemize}
+\item \tcode{C(t())} equals \tcode{C(t())} if and only if
+  \tcode{t()} is an equality preserving
+  expression~(\ref{concepts.lib.general.equality}).
+\item \tcode{C(u())} equals \tcode{C(u())} if and only if
+  \tcode{u()} is an equality preserving
+  expression~(\ref{concepts.lib.general.equality}).
+\end{itemize}
+
+\pnum
+\begin{note}
+Users can customize the behavior of \tcode{Common} by specializing the
+\tcode{common_type} class template\iref{meta.trans.other}.
+\end{note}
+
+\end{itemdescr}
+
+\rSec2[concepts.lib.corelang.integral]{Concept \tcode{Integral}}
```

Comment:
In general, we're trying to avoid single-paragraph sections. This area is getting dangerously close to the cliff, with four subheadings on one page. Is there any plan at all to possibly have multiple (simple) concepts in one subclause? Maybe not. Net result: no change requested for now.

Reply:
I'd be mildly concerned about confusion as to which concepts correspond with which prose requirements - despite that prose requirements are always introduced with "Let `T` be ... Concept<T> is satisfied only if"- but I don't think that's insurmountable.

Reply:
I expect these sections to shrink further once we remove the unnecessary "need be no subsumption" wording.